src/social-network-provider/twitter.com/ui/index.ts

```diff
@@ -41,4 +44,49 @@ export const instanceOfTwitterUI = defineSocialNetworkUI({
         setStorage(twitterUrl.hostIdentifier, { userIgnoredWelcome: true, forceDisplayWelcome: false }).then()
     },
     shouldDisplayWelcome: shouldDisplayWelcomeDefault,
+    useColorScheme() {
+        const [currentScheme, setScheme] = React.useState<'light' | 'dark'>('light')
+        React.useLayoutEffect(() => {
+            const id = setInterval(() => setScheme(isDarkMode()), 2000)
+            return () => clearInterval(id)
+        })
+        return currentScheme
+    },
+    darkTheme: createMuiTheme({
+        ...MaskbookDarkTheme,
+        palette: {
+            ...MaskbookDarkTheme.palette,
+            background: {
+                ...MaskbookDarkTheme.palette.background,
+                get paper() {
+                    return `rgb(${getBackgroundColor().join(',')})`
+                },
+            },
+        },
+    }),
 })
+
+function isDarkMode(): 'dark' | 'light' {
+    const [r, g, b] = getBackgroundColor()
+    if (r < 68 && g < 68 && b < 68) return 'dark'
```

Comment:
This is some crazy color-theory thing going on here.

Reply:
Neruthus tells me this magic value

Reply:
Magic number indeed.